pkg/collector/corechecks/snmp/report_metrics.go

```diff
@@ -0,0 +1,91 @@
+package snmp
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type metricSender struct {
+	sender aggregator.Sender
+}
+
+func (ms *metricSender) reportMetrics(metrics []metricsConfig, metricTags []metricTagConfig, values *snmpValues, tags []string) {
```

Comment:
I would split this in 
* enrichMetrics --> adding tags to all resolved metrics
* [submit|report]Metrics --> calls `sendMetric` for each enriched metric

Reply:
> enrichMetrics --> adding tags to all resolved metrics

Sure, refactored into helper function `getGlobalTags`

> [submit|report]Metrics --> calls sendMetric for each enriched metric

I think it's fine to keep this `reportMetrics` logic in `reportMetrics` since the function is currently not complex. Extracting to `sendMetric` will add some indirection which doesn't necessarily help readability.

```
	for _, metric := range metrics {
		if metric.Symbol.OID != "" {
			ms.reportScalarMetrics(metric, values, newTags)
		} else if metric.Table.OID != "" {
			ms.reportColumnMetrics(metric, values, newTags)
		}
	}
```

Reply:
@pducolin I'm closing this discussion, but feel free to reopen it if needed.